extendedreport.php

```diff
@@ -102,15 +102,70 @@ function extendedreport_civicrm_tabset($tabsetName, &$tabs, $context) {
   foreach ($reports['values'] as $report) {
     $tabs['report_' . $report['id']] = array(
       'title' => ts($report['title']),
-      'url' => CRM_Utils_System::url( 'civicrm/report/contact/addresshistory', array(
+      'url' => CRM_Utils_System::url( 'civicrm/report/instance/' . $report['id'], array(
           'log_civicrm_address_op' => 'in',
           'contact_id_value' => $context['contact_id'],
+          'contact_id' => $context['contact_id'],
           'output' => 'html',
           'force' => 1,
           'section' => 2,
+          'weight' => 70,
         )
       )
     );
   }
 
 }
+
+function extendedreport_civicrm_pageRun(&$page) {
+
+  if (get_class($page) === 'CRM_Contact_Page_View_Summary') {
+    if (($contactID = $page->getVar('_contactId')) !== FALSE) {
+      $page->assign('apiOptions', ['metadata' => ['labels']]);
+      $page->assign('contactID', $contactID);
+    }
+  }
+}
+
+/**
+ * @param $op
+ * @param $objectName
+ * @param $objectId
+ * @param $objectRef
+ */
+function extendedreport__civicrm_post($op, $objectName, $objectId, &$objectRef) {
+  if ($objectName === 'ReportInstance') {
+    CRM_Extendedreport_Page_Inline_ExtendedReportlets::flushReports();
+  }
+}
+
+/**
+ * Implements hook_civicrm_contactSummaryBlocks().
+ *
+ * @link https://github.com/civicrm/org.civicrm.contactlayout
+ */
+function extendedreport_civicrm_contactSummaryBlocks(&$blocks) {
+
+  $reports = CRM_Extendedreport_Page_Inline_ExtendedReportlets::getReportsToDisplay();
+
+  if (empty($reports)) {
+    return;
+  }
+  // Provide our own group for this block to visually distinguish it on the contact summary editor palette.
+  $blocks += [
+    'extendedreports' => [
+      'title' => ts('Extended report dashlets'),
+      'icon' => 'fa-user-circle',
+      'blocks' => [],
+    ]
+  ];
+  foreach ($reports as $report) {
+    $blocks['extendedreports']['blocks'][$report['id']] = [
```

Comment:
I'd suggest making prefixing the id with "report_", so this line would be 

    $blocks['extendedreports']['blocks']['report_' . $report['id']] = [